### ISSUE TITLE
java.util.NoSuchElementException: head of empty list

### ISSUE THREAD

**Issue by baptistejamin** [metarank/metarank#861]
Hello!

I am trying importing data from a custom dataset using `metarank/metarank:latest standalone`.

I am getting the following error:

```
java.util.NoSuchElementException: head of empty list
	at scala.collection.immutable.Nil$.head(List.scala:662)
	at scala.collection.immutable.Nil$.head(List.scala:661)
	at io.github.metarank.ltrlib.model.Query$.apply(Query.scala:20)
	at ai.metarank.flow.ClickthroughQuery$.apply(ClickthroughQuery.scala:25)
	at ai.metarank.ml.rank.LambdaMARTRanker$LambdaMARTPredictor.$anonfun$loadDataset$3(LambdaMARTRanker.scala:127)
	at fs2.Chunk.$anonfun$map$1(Chunk.scala:193)
	at fs2.Chunk.$anonfun$map$1$adapted(Chunk.scala:193)
	at fs2.Chunk.foreachWithIndex(Chunk.scala:160)
	at fs2.Chunk.map(Chunk.scala:193)
	at fs2.Pull$.$anonfun$mapOutputNoScope$1(Pull.scala:1307)
	at fs2.Pull$$anon$1.cont(Pull.scala:149)
	at fs2.Pull$.fs2$Pull$$bindBindAux(Pull.scala:714)
	at fs2.Pull$BindBind.cont(Pull.scala:701)
	at fs2.Pull$ContP.apply(Pull.scala:649)
	at fs2.Pull$ContP.apply$(Pull.scala:649)
	at fs2.Pull$Bind.apply(Pull.scala:657)
	at fs2.Pull$Bind.apply(Pull.scala:657)
	at fs2.Pull$UnconsRunR$1.$anonfun$out$2(Pull.scala:997)
	at fs2.Pull$.$anonfun$compile$2(Pull.scala:930)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Pull$.$anonfun$compile$21(Pull.scala:1182)
	at update @ ai.metarank.flow.MetarankFlow$.$anonfun$process$7(MetarankFlow.scala:28)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at update @ ai.metarank.flow.MetarankFlow$.$anonfun$process$7(MetarankFlow.scala:28)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at update @ ai.metarank.flow.MetarankFlow$.$anonfun$process$7(MetarankFlow.scala:28)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	```

**Comment by vgoloviznin** [comment]
Hey @baptistejamin when running the validation command (https://docs.metarank.ai/reference/cli#validation), what do you see in the logs?

**Comment by baptistejamin** [comment]
They there.

First of all, thank you for building such a thing. The idea behind this is brilliant, and it brings great innovations for smaller companies.

Ask you requested here is the output:

```
+ OPTS='-Xmx1g -verbose:gc'
+ exec /usr/bin/java -Xmx1g -verbose:gc -jar /app/metarank.jar validate --data /opt/metarank/articles2.jsonl --config /opt/metarank/config.yml
[0.004s][info][gc] Using G1
[0.306s][info][gc] GC(0) Pause Young (Normal) (G1 Evacuation Pause) 39M->15M(154M) 2.562ms
[0.533s][info][gc] GC(1) Pause Young (Normal) (G1 Evacuation Pause) 36M->16M(154M) 2.504ms
16:47:54.874 INFO  ai.metarank.main.Main$ - Metarank v0.6.0 is starting.
[0.691s][info][gc] GC(2) Pause Young (Concurrent Start) (Metadata GC Threshold) 36M->17M(154M) 2.047ms
[0.691s][info][gc] GC(3) Concurrent Mark Cycle
[0.696s][info][gc] GC(3) Pause Remark 18M->18M(70M) 1.230ms
[0.701s][info][gc] GC(3) Pause Cleanup 18M->18M(70M) 0.062ms
[0.702s][info][gc] GC(3) Concurrent Mark Cycle 11.602ms
16:47:55.123 INFO  ai.metarank.config.Config$ - api conf block is not defined: using default ApiConfig(Hostname(0.0.0.0),Port(8080))
16:47:55.123 INFO  ai.metarank.config.Config$ - state conf block is not defined: using default MemoryStateConfig()
16:47:55.124 INFO  ai.metarank.config.Config$ - train conf block is not defined: using default MemoryTrainConfig()
16:47:55.214 INFO  ai.metarank.config.Config$ - Loaded config file, state=memory, features=[click,paid,profile,liked_tags,liked_category,liked_author,visitor_click_count,global_item_click_count,day_item_click_count,time,position], models=[similar,trending,xgboost]
16:47:55.233 INFO  ai.metarank.main.Main$ - usage analytics enabled: release=true analytics=true errors=true
[1.026s][info][gc] GC(4) Pause Young (Prepare Mixed) (G1 Evacuation Pause) 57M->18M(70M) 4.526ms
[1.037s][info][gc] GC(5) Pause Young (Mixed) (G1 Evacuation Pause) 19M->19M(70M) 2.901ms
[1.153s][info][gc] GC(6) Pause Young (Normal) (G1 Evacuation Pause) 55M->18M(70M) 1.138ms
[1.188s][info][gc] GC(7) Pause Young (Normal) (G1 Evacuation Pause) 56M->18M(70M) 1.103ms
[1.219s][info][gc] GC(8) Pause Young (Normal) (G1 Evacuation Pause) 56M->18M(129M) 1.760ms
[1.319s][info][gc] GC(9) Pause Young (Concurrent Start) (GCLocker Initiated GC) 88M->19M(129M) 1.279ms
[1.319s][info][gc] GC(10) Concurrent Mark Cycle
[1.326s][info][gc] GC(10) Pause Remark 20M->20M(77M) 1.358ms
[1.332s][info][gc] GC(10) Pause Cleanup 22M->22M(77M) 0.031ms
[1.334s][info][gc] GC(10) Concurrent Mark Cycle 15.003ms
16:47:55.580 INFO  ai.metarank.FeatureMapping - optimized schema: removed 5 unused features
16:47:55.610 INFO  ai.metarank.main.command.Validate$ - Dataset validation is enabled
16:47:55.610 INFO  ai.metarank.main.command.Validate$ - Validation loads all events to RAM, so use --validation=false to skip in case of OOM
16:47:55.627 INFO  ai.metarank.source.FileEventSource - path=/opt/metarank/articles2.jsonl is a file
16:47:55.640 INFO  ai.metarank.source.FileEventSource - file /opt/metarank/articles2.jsonl selected=true (timeMatch=true formatMatch=true)
16:47:55.641 INFO  ai.metarank.source.FileEventSource - reading file /opt/metarank/articles2.jsonl (no compression)
[1.538s][info][gc] GC(11) Pause Young (Prepare Mixed) (G1 Preventive Collection) 73M->31M(85M) 8.692ms
[1.549s][info][gc] GC(12) Pause Young (Mixed) (G1 Evacuation Pause) 32M->32M(85M) 8.036ms
16:47:55.804 INFO  ai.metarank.main.command.Validate$ - Validation done
16:47:55.807 INFO  a.m.v.checks.EventOrderValidation$ - Event ordering check = PASS (1202 events sorted by timestamp)
16:47:55.809 INFO  a.m.v.checks.EventTypesValidation$ - event types check = PASS (1193 item events, 4 rankings, 5 interactions)
16:47:55.817 INFO  a.m.v.c.FeatureOverMissingFieldValidation$ - field reference check = PASS (11 features referencing existing 7 event fields)
16:47:55.819 INFO  a.m.v.c.InteractionKeyValidation$ - interaction-ranking join key check = PASS (4 rankings, all interactions reference existing ones)
16:47:55.828 INFO  a.m.v.c.InteractionMetadataValidation$ - Interaction metadata check: PASS
16:47:55.831 INFO  a.m.v.c.InteractionPositionValidation$ - interaction positions check = PASS (int distribution: [1,2]
16:47:55.834 INFO  a.m.v.c.InteractionTypeValidation$ - interaction type check = PASS (5 interactions have known types: Set(click))
16:47:55.834 INFO  ai.metarank.main.Main$ - My job is done, exiting.
```

I just'd like to mention the dataset is very small. There are around 1500 items, and only a few interactions.

**Comment by vgoloviznin** [comment]
Thanks for trying it out! 

Can you try with the latest 0.6.1 release and see if it will work? 

We've fixed an issue in 0.6.0 yesterday

**Comment by baptistejamin** [comment]
Just tested with the 0.6.1 docker version, same issue.

I can email you the dataset and the config.yml

**Comment by vgoloviznin** [comment]
That'd be great! Please use seva@metarank.ai, we will check it most likely tomorrow

**Comment by vgoloviznin** [comment]
You can also join our slack channel for faster communication: https://metarank.ai/slack

**Comment by shuttie** [comment]
All related issues are fixed in 0.6.2. You're welcome to reopen the issue if it persists on a newer version (but hopefully it should be OK)
